WebApp-OpenIDConnect-DotNet/Controllers/SessionController.cs

```diff
@@ -21,18 +22,28 @@ public SessionController(IOptions<AzureAdB2COptions> b2cOptions)
         public AzureAdB2COptions AzureAdB2COptions { get; set; }
 
         [HttpGet]
-        public IActionResult SignIn()
+        public async Task SignIn()
         {
-            return Challenge(
-                new AuthenticationProperties { RedirectUri = "/" }, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
+                OpenIdConnectDefaults.AuthenticationScheme, new AuthenticationProperties { RedirectUri = "/" });
         }
 
         [HttpGet]
-        public IActionResult ResetPassword()
+        public async Task ResetPassword()
         {
-            var properties = new AuthenticationProperties() { RedirectUri = "/" };
+            var properties = new AuthenticationProperties() { RedirectUri = "/"  };
             properties.Items[AzureAdB2COptions.PolicyAuthenticationProperty] = AzureAdB2COptions.ResetPasswordPolicyId;
-            return Challenge(properties, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
```

Comment:
ChallengeBehavior is required for this to work, and this is only available via ChallengeAsync, not Challenge.
@danroth27 , do you have any concerns with this approach?